### PR TITLE
Add Front-End

### LINKED ISSUE #6: Email Escaping
I was thinking about this for quite some time now.
The "To: " field in the SMTP protocol allows a comma separated to send an emial to multiple recipients at once.
I don't think there's a limit to this, so a maliciously crafted parameter could trick the server into sending a registration email to hundreds of people using just a single request.
Likely not what we want, even if this wouldn't even have a drawback for those people.
I haven't really found a definite answer to the question how one would escape such an input, MARTI simply splits an input string at whitespaces, not more and merges the resulting arrays afterwards, as If it's designed for more than 2 emails.
Ideas?

@ssoloff Unrelated to that:
I'm mostly done with the "backend" for now, it would be nice if you could have an in-depth look at the logic and escaping of user-input once I documented how the routes work, to make sure I haven't accidentally built in any obvious security-realted design flaw.

### PR DESCRIPTION
**Pull request by RoiEXLab**
Closes #6
Fixes a lot of points in #1

TODO after this:
1. Write documentation
2. Write Tests

After that, this should be production-ready

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -1,3 +1,7 @@
 {
-    "extends": "airbnb-base"
-}
\ No newline at end of file
+    "extends": "airbnb-base",
+    "env": {
+        "browser": true,
+        "node": true
```
Comment (ssoloff) [inline .eslintrc.json]: Might want to consider having different ESLint configurations for the back-end and front-end sources so it detects incorrect Node usage in front-end code and vice versa.

File: public/email-templates/verify-dice.html [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{% layout 'base' %}
+<h1>The dice have been cast!</h1>
+<p>Roll-Time: {{ date }}</p>
+<p>Results: {{ dice }}</p>
+<br>
+<p>You can click below to verify your results:</p>
+<a class="big" href="{{ url }}">Verify!</a>
+<br>
+<p>Note we might not be able to correctly verify integrity for very old (1+ years) dice rolls</p>
```
Comment (ssoloff) [inline public/email-templates/verify-dice.html]: Missing period at end of sentence?

File: src/api/api.js [hunk 3]
```diff
@@ -11,6 +11,10 @@ class Api {
     this.validator = new Validator();
   }
 
+  static isSingleEmail(email) {
+    return !/[,\s<>]/.test(email);
```
Comment (ssoloff) [inline src/api/api.js]: There appear to be a few modules out there that validate email address syntax according to the RFC.  Would it be more appropriate to sanity check that the address syntax is truly valid before we attempt to send an email?

File: .eslintrc.json (hunk 1 above)
Comment (RoiEXLab) [reply .eslintrc.json]: Probably the best option, I was just unsure on how to achieve that. I'll have a look

File: src/api/api.js (hunk 3 above)
Comment (RoiEXLab) [reply src/api/api.js]: Yeah there's some controversy around this topic.
There are some super long regexes that are supposed to validate an email correctly, which are almost never used because there are shorter regexes that are supposed to be around 99.9% accurate.
Then again, there are some people that have the opinion that only sending out an email will truly validate that email + will keep up with potentially changing standards.
That's why I chose the simplest route.

Comment (RoiEXLab) [reply src/api/api.js]: Found this site: https://emailregex.com
Will use the proposed JavaScript regex

Comment (ssoloff) [reply src/api/api.js]: I should have clarified my concern better.  I was thinking of the non-malicious case where a user simply entered a malformed email address and had to wait for us to actually try to send the email before receiving an error message indicating the address is invalid.  I forgot to check the client side, and now see that you're using an `email` input element on the form.  That should handle client validation sufficiently to address the scenario I had in my mind. :+1:

But a better regex is still an improvement for our (future) REST API users. :smile:

File: public/.eslintrc.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+    "extends": "airbnb-base",
```
Comment (ssoloff) [inline public/.eslintrc.json:2]: Just for future reference...  ESLint has a [cascading and hierarchical model](https://eslint.org/docs/user-guide/configuring#configuration-cascading-and-hierarchy) for configurations.  So, if in the future, you want to share a more complicated configuration between back-end and front-end code, it should Just Work in the current project layout.

**TL;DR:** This line shouldn't be necessary as it should be picked up from the parent configuration (unless there's something special about the `extends` property I don't know about).

**Review by ssoloff — APPROVED** [review]
